hyp3_metadata/__main__.py

```diff
@@ -0,0 +1,207 @@
+                     plugin_name: str, plugin_version: str, processor_name: str, processor_version: str) -> dict:
+    payload = locals()
+    payload['metadata_version'] = __version__
+
+    payload.update(decode_product(product_dir))
+
+    payload['granule_type'], payload['granule_description'] = get_granule_type(granule_name)
+
+    payload['dem_resolution'] = get_dem_resolution(dem_name)
+
+    return payload
+
+
+def create_readme(payload: dict) -> Path:
+    payload = copy.deepcopy(payload)
+
+    reference_file = payload['product_dir'] / f'{payload["product_dir"].name}.png'  # FIXME: use product(s)?
+
+    return create(payload, 'readme.md.txt.j2', reference_file, out_ext='README.md.txt',
+                  strip_ext=True, thumbnail=False)
+
+
+def create_product_xmls(payload: dict) -> List[Path]:
+    payload = copy.deepcopy(payload)
+
+    output_files = []
+    for pol in get_polarizations(payload['polarizations']):
+        payload['pol'] = pol
+        reference_file = payload['product_dir'] / f'{payload["product_dir"].name}_{pol}.tif'
+
+        output_files.append(
+            create(payload, 'product.xml.j2', reference_file)
+        )
+
+    return output_files
+
+
+def create_dem_xml(payload: dict) -> Path:
+    payload = copy.deepcopy(payload)
+    reference_file = payload['product_dir'] / f'{payload["product_dir"].name}_dem.tif'
+
+    dem_template_id = get_dem_template_id(payload['dem_name'])
+
+    return create(payload, f'dem/dem-{dem_template_id}.xml.j2', reference_file)
+
+
+def create_browse_xml(payload: dict) -> Path:
+    payload = copy.deepcopy(payload)
+    reference_file = payload['product_dir'] / f'{payload["product_dir"].name}.png'
+
+    if reference_file.name.endswith('_rgb.png'):
+        browse_scale = 'color'
+    else:
+        browse_scale = 'greyscale'
+
+    return create(payload, f'browse/browse-{browse_scale}.xml.j2', reference_file)
+
+
+def create_inc_map_xml(payload: dict) -> Path:
+    payload = copy.deepcopy(payload)
+    reference_file = payload['product_dir'] / f'{payload["product_dir"].name}_inc_map.tif'
+    return create(payload, 'inc_map.xml.j2', reference_file)
+
+
+def create_ls_map_xml(payload: dict) -> Path:
+    payload = copy.deepcopy(payload)
+    reference_file = payload['product_dir'] / f'{payload["product_dir"].name}_ls_map.tif'
+    return create(payload, 'ls_map.xml.j2', reference_file)
+
+
+def create(payload: dict, template: str, reference_file: Path = None, out_ext: str = 'xml',
+           strip_ext: bool = False, thumbnail: bool = True) -> Path:
+    if reference_file:
+        info = gdal.Info(str(reference_file), format='json')
+        payload['pixel_spacing'] = info['geoTransform'][1]
+        payload['projection'] = get_projection(info['coordinateSystem']['wkt'])
+
+    if reference_file and thumbnail:
+        payload['thumbnail_binary_string'] = b''  # TODO
```

Comment:
Still outstanding.  For rtc gamma, there will already be a browse image for each of the product files sitting in the product directory.  Currently implementation at https://github.com/ASFHyP3/hyp3-rtc-gamma/blob/develop/hyp3_rtc_gamma/create_metadata.py#L112 uses `hyp3lib.make_arc_thumb.pngtothumb`.  And the current implementation always uses a white thumb for the `ls_map` rather than the actual `ls_map`, Heidi couldn't recall why when I asked last week.
```
browse_filename = reference_file + '.png'
thumbnail = create_thumbnail(browse_filename, ...)
thumbnail_brinary_string = get_binary_string(thumbnail)
```